l1-contracts/src/UniFiAVSManager.sol

```diff
@@ -577,5 +782,70 @@ contract UniFiAVSManager is UniFiAVSManagerStorage, IUniFiAVSManager, UUPSUpgrad
         emit DeregistrationDelaySet(oldDelay, newDelay);
     }
 
+    function _validateProofAndGetValidatorKeys(BeaconChainHelperLib.InclusionProof memory proof, bool checkIndex)
+        internal
+        returns (uint64, bytes32)
+    {
+        UniFiAVSStorage storage $ = _getUniFiAVSManagerStorage();
+
+        if (!BeaconChainHelperLib.verifyValidatorExistence(proof)) {
+            revert InvalidValidatorProof(proof.validator[0]);
+        }
+
+        bytes32 storedBlsPubKeyHash = checkIndex ? $.validatorIndexes[proof.validatorIndex] : proof.validator[0];
+        ValidatorData storage validator = $.validators[storedBlsPubKeyHash];
+
+        if (validator.eigenPod != address(0)) {
+            revert InvalidValidatorType();
+        }
+
+        ValidatorRegistrationData storage validatorRegistrationData = $.validatorRegistrations[storedBlsPubKeyHash];
+        if (proof.timestamp < validatorRegistrationData.registeredAt || proof.timestamp > validator.registeredUntil) {
+            revert InvalidValidatorProof(proof.validator[0]);
+        }
+
+        return (validator.index, storedBlsPubKeyHash);
+    }
+
+    /**
+     * @dev Internal function to slash and deregister a validator
+     * @param blsPubKeyHash The BLS public key hash of the validator
+     */
+    function _slashAndDeregisterValidator(bytes32 blsPubKeyHash) internal {
+        UniFiAVSStorage storage $ = _getUniFiAVSManagerStorage();
+        ValidatorData storage validator = $.validators[blsPubKeyHash];
+        address operator = validator.operator;
+
+        uint64 validatorIndex = validator.index;
+        if (validatorIndex == 0) {
+            revert ValidatorNotFound();
+        }
+
+        $.slashedOperators[operator].push(
+            InvalidValidator({ slashingBeneficiary: msg.sender, blsPubKeyHash: blsPubKeyHash })
```

Comment:
slashingBeneficiary naming is bad, how can somebody benefit from being slashed? :D maybe use `slashedOperator`?